tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -1410,5 +1432,243 @@ Status DenseReader::add_extra_offset() {
   return Status::Ok();
 }
 
```

Comment:
Note to reviewer, these were copy paster from reader_base.cc but modified to take into account query condition results. Overflow detection was updated to check that buffers are big enough from the start instead of during the copy operation.